packages/medusa/src/services/product-variant-inventory.ts

```diff
@@ -311,11 +311,11 @@ class ProductVariantInventoryService extends TransactionBaseService {
 
   /**
    * Remove a variant from an inventory item
-   * @param variantId variant id
+   * @param variantId variant id or undefined if all the variants will be affected
    * @param inventoryItemId inventory item id
    */
   async detachInventoryItem(
-    variantId: string,
+    variantId: string | undefined,
```

Comment:
nit: don't know of it is just me but explicitly using undefined between args feels a little dirty - think any of the following would be cleaner:

**Option 1**
- `detachInventoryItem(invItemId)` - for all
- `detachInventoryItem(invItemId, variantId)` - for speciffic

**Option 2**
- `detachInventoryItem(invItemId)` - for all
- `detachInventoryItemForVariant(invItemId, variantId)` - for specific

**Option 3**
- `detachInventoryItem({ inventoryItemId })` - for all
- `detachInventoryItem({ inventoryItemId, variantId })` - for specific


Non-blocking though so up to you

Reply:
Nice catch, this methods weren't being used, so it was silly to allow undefined here..
Just inverted their parameters and marked it as optional.